src/composition.jl

```diff
@@ -108,25 +108,67 @@ function A_mul_B!(y::AbstractVector, A::CompositeMap, x::AbstractVector)
         A_mul_B!(y, A.maps[1], x)
     else
         T = eltype(y)
-        dest = Array{T}(undef, size(A.maps[1], 1))
```

Comment:
The reason for assigning `dest` this way is to make sure that the `eltype` of `dest` is fixed (to that of `y`) such that it can hopefully be reused later on if there are more than 2 linear maps.  This might fail with the new lines of code below.